frontend/src/components/Meal/Day.js

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { DateTime } from "luxon";
+import { APP_URL } from "../../config";
+import Box from "@material-ui/core/Box";
+import Collapse from "@material-ui/core/Collapse";
+import IconButton from "@material-ui/core/IconButton";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableRow from "@material-ui/core/TableRow";
+import Paper from "@material-ui/core/Paper";
+import Card from "@material-ui/core/Card";
+import CardHeader from "@material-ui/core/CardHeader";
+import CardContent from "@material-ui/core/CardContent";
+import KeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
+import KeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
+import { getBackendAuthToken } from "../../session/localStorage";
+import { useDayStyles } from "../../styles/theme";
+
+function getCardSubtitleAsCalories(meals) {
+  var total = 0;
+  var i;
+  for (i = 0; i < meals.length; i++) {
```

Comment:
This whole function can be written like that:

```
const getCardSubtitleAsCalories = meals =>`${meals.reduce((prev, current) => prev + current.total_kcal, 0)} kcal`
```

it's perfect usage case for javascript reduce https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

I would also change the name of this function, it's bit weird and does not explain much. Maybe just `mealsTotalKcalSum` instead? And make it return int, and concatenate a string where function is invoked.

Reply:
OK